scipy/signal/filter_design.py

```diff
@@ -683,16 +683,16 @@ def group_delay(system, w=512, whole=False, fs=2*pi):
     z = np.exp(-1j * w)
     num = np.polyval(cr[::-1], z)
     den = np.polyval(c[::-1], z)
-    singular = np.absolute(den) < 10 * EPSILON
+
+    gd = np.real(num / den) - a.size + 1
+    singular = ~np.isfinite(gd)
     if np.any(singular):
+        gd[singular] = 0
```

Comment:
The CI failures are real--obviously related to code shifting around the warning below.

It looks like there's a pretty detailed discussion in the matching issue so let me ping: @e-q @josephernest and @ewmoore from that thread for feedback here.

Reply:
Sorry about that, I should have looked at the test before pushing the pull request. I'm not sure the test for singularity is still relevant though.

in theory, you can get infinite from numpy which can be tested for example by `np.divide([1],[0])` which yields `array([inf])`, which is why I think it's good to keep the safeguard to ensure we're returning valid numbers.

But in practice for filter design I suspect that because of the numerical approximations going on, it is almost impossible to evaluate the group delay at the exact frequency where it would become singular. Instead, as we can see from the test, we get very large numbers as a results, but valid numbers nonetheless.

Current test:
```python
    z1 = np.exp(1j * 0.1 * pi)
    z2 = np.exp(1j * 0.25 * pi)
    p1 = np.exp(1j * 0.5 * pi)
    p2 = np.exp(1j * 0.8 * pi)
    b = np.convolve([1, -z1], [1, -z2])
    a = np.convolve([1, -p1], [1, -p2])
    w = np.array([0.1, 0.25 * pi, -0.5 * pi, -0.8 * pi])

    w, gd = signal.group_delay((b, a), w=w)
```

yields the following at the "singular" frequencies:
`array([ 7.10542736e-15,  1.06526099e+16, -3.23561693e+15,  3.97059067e+15])`


I don't think getting those very high values in the results is problematic as such, as I think it's better than replacing them with zeros, as it gives continuity when plotting the group delay and also allows finding the point that tends towards infinity by looking at the maximum. 

The previous check also didn't only replace the peak singular values with zeros but a whole band of values that were estimated to be too high, preventing you from evaluating the trend of the group delay in function of frequency.

For example, the following code implements a narrow band 9th order Butterworth low pass filter with a 1000Hz sample rate and 12 Hz cuttoff frequency:
```python
    b, a = signal.butter(9, 12, fs=1000)
    w, gd = signal.group_delay((b, a))
    plt.plot(w, gd)
```

With the *previous check*, all the pass band group delay would be cut off in the results:
![image](https://user-images.githubusercontent.com/7825368/105573904-bd418a80-5d2e-11eb-929a-e869eaf8aab3.png)


With the *new check*, no values are cut off and the trend can be easily interpreted:
![image](https://user-images.githubusercontent.com/7825368/105573923-e3ffc100-5d2e-11eb-9b41-8ff330663960.png)

Instead of cutting off possibly valid results, if someone wants to find the singularity points it is their job to interpret the results and establish a singularity threshold that works for their application. The new singularity check is there only to ensure we return valid numbers in all cases.

So all in all I think either we remove the test case entirely or we find a case where goup_delay will actually generate an infinite value, but I did a few tests and I'm not sure that's feasible.

Reply:
I pushed a new pull request removing the test case.